java/vector/src/main/java/org/apache/arrow/vector/ZeroVector.java

```diff
@@ -17,132 +17,48 @@
 
 package org.apache.arrow.vector;
 
-import static org.apache.arrow.vector.complex.BaseRepeatedValueVector.DATA_VECTOR_NAME;
-
-import java.util.Collections;
-import java.util.Iterator;
-import java.util.List;
-
 import org.apache.arrow.memory.BufferAllocator;
-import org.apache.arrow.memory.OutOfMemoryException;
-import org.apache.arrow.vector.compare.VectorVisitor;
-import org.apache.arrow.vector.complex.impl.NullReader;
-import org.apache.arrow.vector.complex.reader.FieldReader;
-import org.apache.arrow.vector.ipc.message.ArrowFieldNode;
-import org.apache.arrow.vector.types.Types.MinorType;
-import org.apache.arrow.vector.types.pojo.ArrowType.Null;
-import org.apache.arrow.vector.types.pojo.Field;
-import org.apache.arrow.vector.types.pojo.FieldType;
 import org.apache.arrow.vector.util.CallBack;
 import org.apache.arrow.vector.util.TransferPair;
 
-import io.netty.buffer.ArrowBuf;
-
 /**
  * A zero length vector of any type.
  */
-public final class ZeroVector implements FieldVector {
+public final class ZeroVector extends NullVector {
   public static final ZeroVector INSTANCE = new ZeroVector();
 
-  private final TransferPair defaultPair = new TransferPair() {
-    @Override
-    public void transfer() {
-    }
-
-    @Override
-    public void splitAndTransfer(int startIndex, int length) {
-    }
-
-    @Override
-    public ValueVector getTo() {
-      return ZeroVector.this;
-    }
-
-    @Override
-    public void copyValueSafe(int from, int to) {
-    }
-  };
-
-
   public ZeroVector() {
   }
 
   @Override
-  public void close() {
+  public int getValueCount() {
+    return 0;
   }
 
   @Override
-  public void clear() {
+  public void setValueCount(int valueCount) {
```

Comment:
if valueCount > 0 should it be an error?

Reply:
I prefer not add check here since it dose nothing here and more important this will cause some unexpected exception which works fine before (i.e. if we create a ListVector and setValueCount it will fail since it's initialized with ZeroVector)